api/proto/teleport/legacy/types/types.proto

```diff
@@ -6675,6 +6675,10 @@ message PluginOktaSyncSettings {
   // This is useful when the app groups are not needed in Teleport.
   // and integration with Okta is only used for user sync.
   bool disable_sync_app_groups = 9;
+
+  // DisableBidirectionalSync prevents syncing anything from Teleport to Okta. I.e. prevents
+  // creating Okta assignments.
+  bool disable_bidirectional_sync = 10;
```

Comment:
Should we consider a non-boolean field, so it could deal with different sync directions?

Alternatively should we make it clear, in the field name, that it disables only Teleport-to-Okta? Eg, disable_teleport_to_okta_sync?

- https://protobuf.dev/best-practices/api/#never-use-booleans-for-two-states

Reply:
The field name comes from the toggle name in the UI design, but I understand where you're coming from. @smallinsky do you have an opinion here?

Reply:
I don’t think this will extend in the future. We plan to support sync only from Okta to Teleport , and reversed  Teleport to Okta scenario doesn’t make sense from a product perspective.

Reply:
> I don’t think this will extend in the future

Famous last words.  ;)

I won't push this further if you think it's the way to go, but please consider a non-boolean in future API changes.

Reply:
We can consider to squash all existing bool flags into a separate enums Okta (like `okta sync mode`) mode but adding `disable_bidirectional_sync` will not block this and we can add it later but for taht we will need do a serious refactoring.